### PR TITLE
[GEOS-11654] Fix multiline strings that are missing a space between the lines

### PR DESCRIPTION
**Pull request by petersmythe**
https://osgeo-org.atlassian.net/browse/GEOS-11654

Multiline strings where the second line starts with a character, should end with a space in the previous line:

e.g. 
```java
            throw new IllegalArgumentException(
                    "Requested bounding box contains wrong"
                            + "number of coordinates (should have "
...
```
produces `<ows:ExceptionText>java.lang.IllegalArgumentException: Requested bounding box contains wrongnumber of coordinates (should have 4): 0
Requested bounding box contains wrongnumber of coordinates (should have 4): 0</ows:ExceptionText>`

I picked up most with `find . -type f -exec awk '/[^\\][a-zA-Z]"$/ { getline nextline; if (nextline ~ /^[ \t]*\+\s"[a-zA-Z]/) {print FILENAME, $0 "\n" nextline }}' {} +` 

i.e. a character (not preceded by a backslash \) then quotes then newline then whitespace then plus then space then quotes then a character

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

File: src/community/features-templating/features-templating-core/src/main/java/org/geoserver/featurestemplating/request/TemplatePathVisitor.java [hunk 1]
```diff
@@ -90,7 +90,7 @@ protected Expression mapPropertyThroughBuilder(String propertyName, TemplateBuil
             }
         } catch (Throwable ex) {
             throw new RuntimeException(
-                    "Unable to evaluate template path against"
+                    "Unable to evaluate template path against "
                             + "the template. Cause: "
```
Comment (mprins) [inline src/community/features-templating/features-templating-core/src/main/java/org/geoserver/featurestemplating/request/TemplatePathVisitor.java:94]:
```suggestion
                    "Unable to evaluate template path against the template. Cause: "
```

File: src/extension/ogcapi/ogcapi-core/src/main/java/org/geoserver/ogcapi/APIBBoxParser.java [hunk 2]
```diff
@@ -145,7 +145,7 @@ public static ReferencedEnvelope[] parse(String value, CoordinateReferenceSystem
         // check to make sure that the bounding box has 4 coordinates
         if (unparsed.size() < 4) {
             throw new IllegalArgumentException(
-                    "Requested bounding box contains wrong"
+                    "Requested bounding box contains wrong "
                             + "number of coordinates (should have "
                             + "4): "
```
Comment (mprins) [inline src/extension/ogcapi/ogcapi-core/src/main/java/org/geoserver/ogcapi/APIBBoxParser.java:150]:
```suggestion
                    "Requested bounding box contains wrong number of coordinates (should have 4): "
```

File: src/main/src/main/java/org/geoserver/catalog/CoverageViewReader.java [hunk 3]
```diff
@@ -658,7 +658,7 @@ private ColorModel getColorModelWithAlpha(int currentBandCount) {
                     cs, nBits, true, false, Transparency.TRANSLUCENT, DataBuffer.TYPE_BYTE);
         } else {
             throw new IllegalArgumentException(
-                    "Cannot create a color model with alpha"
+                    "Cannot create a color model with alpha "
                             + "support starting with "
```
Comment (mprins) [inline src/main/src/main/java/org/geoserver/catalog/CoverageViewReader.java:662]:
```suggestion
                    "Cannot create a color model with alpha support starting with "
```

File: src/wcs1_0/src/main/java/org/geoserver/wcs/kvp/BBoxKvpParser.java [hunk 4]
```diff
@@ -32,7 +32,7 @@ public GeneralBounds parse(String value) throws Exception {
         // check to make sure that the bounding box has 4 coordinates
         if (unparsed.size() != 4) {
             throw new WcsException(
-                    "Requested bounding box contains wrong"
+                    "Requested bounding box contains wrong "
                             + "number of coordinates: "
```
Comment (mprins) [inline src/wcs1_0/src/main/java/org/geoserver/wcs/kvp/BBoxKvpParser.java:36]:
```suggestion
                    "Requested bounding box contains wrong number of coordinates: "
```

File: src/wcs1_1/src/main/java/org/geoserver/wcs/kvp/BoundingBoxKvpParser.java [hunk 5]
```diff
@@ -36,7 +36,7 @@ public Object parse(String value) throws Exception {
         // check to make sure that the bounding box has 4 coordinates
         if (unparsed.size() < 4) {
             throw new WcsException(
-                    "Requested bounding box contains wrong"
+                    "Requested bounding box contains wrong "
                             + "number of coordinates (should have at least 4): "
```
Comment (mprins) [inline src/wcs1_1/src/main/java/org/geoserver/wcs/kvp/BoundingBoxKvpParser.java:40]:
```suggestion
                    "Requested bounding box contains wrong number of coordinates (should have at least 4): "
```

File: src/wfs/src/main/java/org/geoserver/wfs/kvp/BBoxKvpParser.java [hunk 6]
```diff
@@ -30,7 +30,7 @@ public Object parse(String value) throws Exception {
         // check to make sure that the bounding box has 4 coordinates
         if (unparsed.size() < 4) {
             throw new IllegalArgumentException(
-                    "Requested bounding box contains wrong"
+                    "Requested bounding box contains wrong "
                             + "number of coordinates (should have "
                             + "4): "
```
Comment (mprins) [inline src/wfs/src/main/java/org/geoserver/wfs/kvp/BBoxKvpParser.java:35]:
```suggestion
                    "Requested bounding box contains wrong number of coordinates (should have 4): "
```

**Review by mprins — COMMENTED** [review]
I would prefer a line that is a little longer and not broken into little pieces when possible, also since we're talking about extending line length/columns with Palantir formatter: https://github.com/geoserver/geoserver/pull/7978 I would not be too strict on the current limit

**Comment by petersmythe** [comment]
I wasn't going to go that far, but I'll gladly accept your suggestion @mprins; hoping that the current spotless formatter does not complain that they are now too long.

**Review by mprins — APPROVED** [review]

**Pull request merged**